src/FormHandler/HandlerBuilder.php

```
@@ -0,0 +1,99 @@
+<?php
+namespace Hostnet\Component\FormHandler;
+
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+
+final class HandlerBuilder implements HandlerConfigInterface
+{
+    private $type = FormType::class;
+    private $name;
+    private $options = [];
+    private $on_success;
+    private $on_failure;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setType($type)
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setName($name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setOptions($options)
+    {
+        $this->options = $options;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function onSuccess(callable $callback)
+    {
+        $this->on_success = $callback;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function onFailure(callable $callback)
+    {
+        $this->on_failure = $callback;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function registerActionSubscriber(ActionSubscriberInterface $action_subscriber)
+    {
+        $actions = array_merge(
+            [HandlerActions::SUCCESS => null, HandlerActions::FAILURE => null],
+            $action_subscriber->getSubscribedActions()
```

You should throw a `new UnknownSubscribedActionException` if something else is asked.